h2/src/main/org/h2/command/dml/Select.java

```diff
@@ -380,6 +431,10 @@ private void queryGroup(int columnCount, LocalResult result) {
             row = keepOnlyDistinct(row, columnCount);
             result.addRow(row);
         }
+        groupByData = null;
```

Comment:
May be we need to clear `groupByData` in finally block? I'm not sure what happens in case of exception in the middle.